syntax = "proto3";

import "common.proto";

message TgAuthRequest {
  string uid = 1;
  string phone = 2;
  string code = 3;
  string code_hash = 4;
}

service TgApi {
  rpc auth(TgAuthRequest) returns (AuthResponse) {};
  rpc get_dialogs(User) returns (Dialogs) {};
  rpc get_messages(MessagesRequest) returns (Messages) {};
  rpc send_message(Send) returns (StatusMessage) {};
  rpc test_file(Text) returns (stream Chunk) {};
}
